File: invocacionWSVC/invocacionWSVC/Form1.vb
Imports System.Data.SqlClient
Imports System.Data
Imports System.Data.Sql
Public Class Form1
    Dim consumirSP As New WSVC.servicioVCSoapClient
    Dim nombre, apePaterno, apeMaterno, nivel As String
    Dim edad, id As Integer
    Dim estatus As Boolean
    Dim conjunto As New DataSet
    Dim datos As New SqlDataAdapter
    
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles btnGuardarSP.Click
        nombre = txtNombre.Text
        apePaterno = txtPaterno.Text
        apeMaterno = txtMaterno.Text
        edad = txtEdad.Text
        estatus = ckbEstatus.Checked
        nivel = cmbNivel.SelectedItem.ToString()
        If (consumirSP.deportistaGuardarSP(nombre, apePaterno, apeMaterno, edad, estatus, nivel) = True) Then
            MessageBox.Show("guardado")
        Else
            MessageBox.Show("Error al guardar")
        End If
    End Sub

    Private Sub btnBuscarTodosSP_Click(sender As Object, e As EventArgs) Handles btnBuscarTodosSP.Click
        conjunto = consumirSP.deprtistaBuscarTodosSP()
        If (conjunto.Tables(0).Rows.Count > 0) Then
            dgvDepostistas.DataSource = conjunto.Tables("DatosGenerales")
        Else
            MessageBox.Show("No existen registros")
        End If
    End Sub

    Private Sub btnBuscarIDSP_Click(sender As Object, e As EventArgs) Handles btnBuscarIDSP.Click
        id = txtID.Text
        conjunto = consumirSP.deportistaBuscarIDSP(id)
        dgvDepostistas.DataSource = conjunto.Tables("DatosGenerales")
        'bajamos los datos al grid
        If (conjunto.Tables(0).Rows.Count > 0) Then
            txtNombre.Text = conjunto.Tables(0).Rows(0)(1).ToString()
            txtPaterno.Text = conjunto.Tables(0).Rows(0)(2).ToString()
            txtMaterno.Text = conjunto.Tables(0).Rows(0)(3).ToString()
            txtEdad.Text = conjunto.Tables(0).Rows(0)(4).ToString()
            ckbEstatus.Checked = conjunto.Tables(0).Rows(0)(5).ToString()
            cmbNivel.SelectedItem = conjunto.Tables(0).Rows(0)(6).ToString()
        Else
            MessageBox.Show("No tenemos registros")
        End If
    End Sub

    Private Sub btnEliminarSP_Click(sender As Object, e As EventArgs) Handles btnEliminarSP.Click
        id = txtID.Text
        If (consumirSP.deportistaEliminarSP(id) = True) Then
            MessageBox.Show("Registro eliminado correctamente")
        Else
            MessageBox.Show("No se pudo eliminar el registro")
        End If
    End Sub

    Private Sub btnActualizarSP_Click(sender As Object, e As EventArgs) Handles btnActualizarSP.Click
        nombre = txtNombre.Text
        apePaterno = txtPaterno.Text
        apeMaterno = txtMaterno.Text
        edad = txtEdad.Text
        estatus = ckbEstatus.Checked
        nivel = cmbNivel.SelectedItem.ToString()
        id = txtID.Text
        If (consumirSP.deportistaActualizarSP(id, nombre, apePaterno, apeMaterno, edad, estatus, nivel) = True) Then
            MessageBox.Show("Actualizado correctamente")
        Else
            MessageBox.Show("Error al actualizar")
        End If
    End Sub
End Class

File: invocacionWSVB(0)/invocacionWSVB/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
//using System.Threading.Tasks;
using System.Windows.Forms;
//agregar
using System.Data.SqlClient;
using System.Data;
using System.Data.Sql;

namespace invocacionWSVB
{
    public partial class Form1 : Form
    {
        //hacemos llabel1 instancia del servicio con store procedure
        WSVBSP.servicioVBSPSoapClient consumirSP = new WSVBSP.servicioVBSPSoapClient();
        //hacemos la instancia del servicio sin store procedure
        WSVB.servicioVBSoapClient consumirNormal = new
            WSVB.servicioVBSoapClient();
        string nombre, apePaterno, apeMaterno, nivel;
        int edad, id;
        bool estatus;
        DataSet conjunto = new DataSet();
        SqlDataAdapter datos = new SqlDataAdapter();
        public Form1()
        {
            InitializeComponent();
        }
        #region "controles SP"
        private void btnGuardar_Click(object sender, EventArgs e)
        {
            nombre= txtNombre.Text;
            apePaterno=txtApePaterno.Text;
            apeMaterno=txtApeMaterno.Text;
            edad=int.Parse(txtEdad.Text);
            estatus=ckbEstatus.Checked;
            nivel= cmbNivel.SelectedItem.ToString();
            if(consumirSP.deportistaGuardarSP(nombre, apePaterno, apeMaterno, edad, estatus, nivel)==true)
            {
                MessageBox.Show("Guardado");
            }
            else
            {
                MessageBox.Show("Error al guardar");
            }
        }
        private void btnBuscarIDSP_Click_1(object sender, EventArgs e)
        {
            id =int.Parse(txtID.Text);
            conjunto = consumirSP.deportistaBuscarIDSP(id);
            dgvDeportistas.DataSource = conjunto.Tables["DatosGenerales"];
            //bajamos los datos al grid
            if (conjunto.Tables[0].Rows.Count>0)
            {
                txtNombre.Text = conjunto.Tables[0].Rows[0][1].ToString();
                txtApePaterno.Text = conjunto.Tables[0].Rows[0][2].ToString();
                txtApeMaterno.Text = conjunto.Tables[0].Rows[0][3].ToString();
                txtEdad.Text = conjunto.Tables[0].Rows[0][4].ToString();
                ckbEstatus.Checked=bool.Parse(conjunto.Tables[0].Rows[0][5].ToString());
                cmbNivel.SelectedItem = conjunto.Tables[0].Rows[0][6].ToString();
            }
            else
            {
                MessageBox.Show("No tenemos registros");
            }

        } 
        
        private void btnGuardarNormal_Click(object sender, EventArgs e)
        {
            nombre = txtNombre.Text;
            apePaterno = txtApePaterno.Text;
            apeMaterno = txtApeMaterno.Text;
            edad = int.Parse(txtEdad.Text);
            estatus = ckbEstatus.Checked;
            nivel = cmbNivel.SelectedItem.ToString();
            if (consumirNormal.deportistaGuardar(nombre, apePaterno, apeMaterno, edad, estatus, nivel) == true)
            {
                MessageBox.Show("guardado");
            }
            else
            {
                MessageBox.Show("Error al guardar");
            }
        }

        private void btnBuscarTodosSP_Click(object sender, EventArgs e)
        {
            conjunto = consumirSP.deportistaBuscarTodosSP();
            if (conjunto.Tables[0].Rows.Count > 0)
            {
                dgvDeportistas.DataSource = conjunto.Tables["DatosGenerales"];
            }
            else
            {
                MessageBox.Show("No existen registros");
            }
        }

        private void btnEliminarSP_Click(object sender, EventArgs e)
        {
            id = int.Parse(txtID.Text);
            if (consumirSP.deportistaEliminarSP(id) == true)
            {
                MessageBox.Show("Registro eliminado correctamente");
            }
            else
            {
                MessageBox.Show("No se pudo eliminar el registro");
            }
        }

        private void btnActualizarSP_Click(object sender, EventArgs e)
        {
            nombre = txtNombre.Text;
            apePaterno = txtApePaterno.Text;
            apeMaterno = txtApeMaterno.Text;
            edad = int.Parse(txtEdad.Text);
            estatus = ckbEstatus.Checked;
            nivel = cmbNivel.SelectedItem.ToString();
            id = int.Parse(txtID.Text);
            if (consumirSP.deportistaActualizarSP(id,nombre, apePaterno, apeMaterno, edad, estatus,nivel) == true)
            {
                MessageBox.Show("Actualizado correctamente");
            }
            else
            {
                MessageBox.Show("Error al actualizar");
            }
        }

        private void btnBuscarIDNormal_Click(object sender, EventArgs e)
        {
            id = int.Parse(txtID.Text);
            conjunto = consumirSP.deportistaBuscarIDSP(id);
            //bajamos los datos al grid
            if (conjunto.Tables[0].Rows.Count > 0)
            {
                txtNombre.Text = conjunto.Tables[0].Rows[0][1].ToString();
                txtApePaterno.Text = conjunto.Tables[0].Rows[0][2].ToString();
                txtApeMaterno.Text = conjunto.Tables[0].Rows[0][3].ToString();
                txtEdad.Text = conjunto.Tables[0].Rows[0][4].ToString();
                ckbEstatus.Checked = bool.Parse(conjunto.Tables[0].Rows[0][5].ToString());
                cmbNivel.SelectedItem = conjunto.Tables[0].Rows[0][6].ToString();
            }
            else
            {
                MessageBox.Show("No tenemos registros");
            }
        }

        private void btnEliminarNormal_Click(object sender, EventArgs e)
        {
            id = int.Parse(txtID.Text);
            if (consumirSP.deportistaEliminarSP(id) == true)
            {
                MessageBox.Show("Registro eliminado correctamente");
            }
            else
            {
                MessageBox.Show("No se pudo eliminar el registro");
            }
        }

        private void btnBuscarTodosNormal_Click(object sender, EventArgs e)
        {
            conjunto = consumirNormal.deportistaBuscarTodos();
            if (conjunto.Tables[0].Rows.Count > 0)
            {
                dgvDeportistas.DataSource = conjunto.Tables["DatosGenerales"];
            }
            else
            {
                MessageBox.Show("No existen registros");
            }
        }

        private void btnActualizarNormal_Click(object sender, EventArgs e)
        {
            nombre = txtNombre.Text;
            apePaterno = txtApePaterno.Text;
            apeMaterno = txtApeMaterno.Text;
            edad = int.Parse(txtEdad.Text);
            estatus = ckbEstatus.Checked;
            nivel = cmbNivel.SelectedItem.ToString();
            id = int.Parse(txtID.Text);
            if (consumirSP.deportistaActualizarSP(id, nombre, apePaterno, apeMaterno, edad, estatus, nivel) == true)
            {
                MessageBox.Show("Actualizado correctamente");
            }
            else
            {
                MessageBox.Show("Error al actualizar");
            }
        }
        #endregion



    }
}
